mypyc/genops.py

```diff
@@ -335,6 +337,7 @@ def __init__(self, fdef: FuncDef, namespace: str) -> None:
         # function.
         self.is_nested = False
         self.contains_nested = False
+        self.ir = None  # type: Optional[FuncIR]
```

Comment:
I'm not a fan of shoving `ir` and `callable_reg` into `FuncInfo`. FuncInfo is for holding the stack of info we need to compile functions. I think the right thing to do here is to make `gen_func_def` return a `Tuple[FuncIR, Value]`, containing the generated function body and (if applicable) the value of the instantiated function object.

Reply:
Done.